src/Swarm/Game/Robot.hs

```diff
@@ -243,12 +275,16 @@ robotContext :: Lens' Robot RobotContext
 -- | The (unique) ID number of the robot.  This is only a Getter since
 --   the robot ID is immutable.
 robotID :: Getter Robot RID
-robotID = to (runIdentity . _robotID)
 
--- | Set the ID number of a robot, changing it from unidentified to
---   identified.
-setRobotID :: RID -> URobot -> Robot
-setRobotID i r = r {_robotID = Identity i}
+-- | Set the ID number of a robot, changing it from template to
+--   concrete.  If it didn't have a location already, just set the
+--   location to (0,0) by default.
+setRobotID :: RID -> TRobot -> Robot
+setRobotID i r =
+  r
+    { _robotID = i
+    , _robotLocation = fromMaybe (V2 0 0) (_robotLocation r)
+    }
```

Comment:
I would not expect `setRobotID` to set the location, maybe change the name?

Reply:
Good idea. :heavy_check_mark: